cfdm/__init__.py

```diff
@@ -63,93 +64,108 @@
+            f"Got {netCDF4.__version__} at {netCDF4.__file__}"
+        )
 
 # Check the version of h5netcdf
 try:
     import h5netcdf
 except ImportError as error1:
     raise ImportError(_error0 + str(error1))
-
-_minimum_vn = "1.3.0"
-if Version(h5netcdf.__version__) < Version(_minimum_vn):
-    raise ValueError(
-        f"Bad h5netcdf version: cfdm requires h5netcdf>={_minimum_vn}. "
-        f"Got {h5netcdf.__version__} at {h5netcdf.__file__}"
-    )
+else:
+    _minimum_vn = "1.3.0"
+    if Version(h5netcdf.__version__) < Version(_minimum_vn):
+        raise ValueError(
+            f"Bad h5netcdf version: cfdm requires h5netcdf>={_minimum_vn}. "
+            f"Got {h5netcdf.__version__} at {h5netcdf.__file__}"
+        )
 
 # Check the version of h5py
 try:
     import h5py
 except ImportError as error1:
     raise ImportError(_error0 + str(error1))
-
-_minimum_vn = "3.10.0"
-if Version(h5py.__version__) < Version(_minimum_vn):
-    raise ValueError(
-        f"Bad h5py version: cfdm requires h5py>={_minimum_vn}. "
-        f"Got {h5py.__version__} at {h5py.__file__}"
-    )
+else:
+    _minimum_vn = "3.12.0"
+    if Version(h5py.__version__) < Version(_minimum_vn):
+        raise ValueError(
+            f"Bad h5py version: cfdm requires h5py>={_minimum_vn}. "
+            f"Got {h5py.__version__} at {h5py.__file__}"
+        )
 
 # Check the version of s3fs
 try:
     import s3fs
 except ImportError as error1:
     raise ImportError(_error0 + str(error1))
-
-_minimum_vn = "2024.6.0"
-if Version(s3fs.__version__) < Version(_minimum_vn):
-    raise ValueError(
-        f"Bad s3fs version: cfdm requires s3fs>={_minimum_vn}. "
-        f"Got {s3fs.__version__} at {s3fs.__file__}"
-    )
+else:
+    _minimum_vn = "2024.6.0"
+    if Version(s3fs.__version__) < Version(_minimum_vn):
+        raise ValueError(
+            f"Bad s3fs version: cfdm requires s3fs>={_minimum_vn}. "
+            f"Got {s3fs.__version__} at {s3fs.__file__}"
+        )
 
 # Check the version of scipy
 try:
     import scipy
 except ImportError as error1:
     raise ImportError(_error0 + str(error1))
-
-_minimum_vn = "1.10.0"
-if Version(scipy.__version__) < Version(_minimum_vn):
-    raise ValueError(
-        f"Bad scipy version: cfdm requires scipy>={_minimum_vn}. "
-        f"Got {scipy.__version__} at {scipy.__file__}"
-    )
+else:
+    _minimum_vn = "1.10.0"
+    if Version(scipy.__version__) < Version(_minimum_vn):
+        raise ValueError(
+            f"Bad scipy version: cfdm requires scipy>={_minimum_vn}. "
+            f"Got {scipy.__version__} at {scipy.__file__}"
+        )
 
 # Check the version of dask
 try:
     import dask
 except ImportError as error1:
     raise ImportError(_error0 + str(error1))
+else:
+    _minimum_vn = "2024.6.0"
+    _maximum_vn = "2024.7.1"
+    _dask_version = Version(dask.__version__)
+    if _dask_version <= Version(_minimum_vn) or _dask_version >= Version(_maximum_vn):
```

Comment:
My bad on a previous suggestion!

```suggestion
    if _dask_version < Version(_minimum_vn) or _dask_version > Version(_maximum_vn):
```